py/terra_widgets/workspace_metadata.py

```diff
@@ -0,0 +1,92 @@
+"""Methods to obtain workspace metadata for the current user in various formats."""
+
+import json
+import os
+from typing import Dict
+
+import firecloud.api as fapi
+from IPython import get_ipython
+
+
+class WorkspaceMetadata:
+  """Encapsulate all logic for obtaining workspace metadata."""
+
+  AOU_DOMAIN = '@researchallofus.org'
+  EDIT_ACCESS_LEVELS = ['WRITER', 'OWNER', 'PROJECT_OWNER']
+  AOU_PROD_API = 'https://api.workbench.researchallofus.org/v1/workspaces'
+
+  def __init__(self):
+    self.user = os.getenv('OWNER_EMAIL')
+    self.terra_workspaces = fapi.list_workspaces().json()
+    if self.user.endswith(self.AOU_DOMAIN):
+      aou_api = os.getenv('RW_API_BASE_URL')
+      if not aou_api:
+        aou_api = self.AOU_PROD_API
+      # Use the All of Us API to get the human-readable workspace names. For All of Us workspaces,
+      # the Terra workspace metadata the workspace names are actually the AoU workspace ids.
+      aou_response = get_ipython().getoutput(f'''curl -H "Content-Type: application/json" \
+          -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+          "{aou_api}" 2>/dev/null | jq .''')
+      self.aou_workspaces = json.loads(''.join(aou_response))['items']
+    else:
+      self.aou_workspaces = None
+
+  def get_workspace_name_to_id_mapping(self, include_private_readonly: bool = False, include_all: bool = False) -> Dict[str, str]:
```

Comment:
nit: these boolean params seem mutually exclusive, may indicate it should be an enum. Alternatively, perhaps just kill `include_all` since it doesn't seem to be used anywhere.

Reply:
Yes, one is a subset of the other. 

On Terra `include_all` yields a long list of featured workspaces. Its in there because I wanted to make it possible to see all of those even though `display_html_snapshots_widget()` isn't making use of it currently. But I have a feeling workspace filtering is going to go through a few iterations for Terra use cases, so just keeping it simple for now.